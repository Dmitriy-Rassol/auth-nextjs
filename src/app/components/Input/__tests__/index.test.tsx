// @ts-check

import { fireEvent, render, screen } from '@testing-library/react';
import { Input } from '../index';
import '@testing-library/jest-dom';

describe('Input', () => {


    const setup = () => {
        const handler = jest.fn();

        const utils = render(<Input
            type="text"
            label="My Label"
            placeholder="My Placeholder"
            handleChange={handler}
            name=''
        />)
        const input = screen.getByLabelText('My Label') as HTMLInputElement;
        // Assertion, так как мы уверенны, что здесь будет input-элемент,
        // а тип по умолчанию HTMLElement не содержит свойства value, необходимого нам в 3м тесте

        return {
            input,
            handler,
            ...utils,
        }
    }


    it('Should match snapshot', () => {
        const { container } = setup();

        expect(container).toMatchSnapshot();
    })


    it('Should render a label', () => {
        setup();
        const label = screen.getByText('My Label');

        expect(label).toBeInTheDocument();
    });

    // https://testing-library.com/docs/example-input-event/ хороший пример теста другого Input-компонента

    it('Should be changeable and call handler on change', () => {
        const { input, handler } = setup();

        fireEvent.change(input, {
            target: {
                value: 'Some Value'
            }
        })

        expect(input.value).toBe('Some Value');
        expect(handler).toHaveBeenCalled();
        expect(handler).toHaveBeenCalledWith('Some Value');
    })
});