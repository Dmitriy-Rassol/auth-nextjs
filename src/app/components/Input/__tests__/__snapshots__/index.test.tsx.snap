// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should match snapshot 1`] = `
<div>
  <div
    class="mt-8"
  >
    <label
      class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
    >
      My Label
    </label>
    <input
      aria-label="My Label"
      class="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
      id="_My Label"
      name=""
      placeholder="My Placeholder"
      required=""
      type="text"
      value=""
    />
  </div>
</div>
`;
